tst/EnergyPlus/unit/RoomAirflowNetwork.unit.cc

```diff
@@ -580,3 +603,234 @@ TEST_F(EnergyPlusFixture, RoomAirInternalGains_InternalHeatGains_Check)
 
     EXPECT_TRUE(compare_err_stream(error_string, true));
 }
+
+TEST_F(EnergyPlusFixture, RoomAirflowNetwork_CheckEquipName_Test)
+{
+    // Test #6321
+    bool check;
+    std::string const EquipName = "ZoneEquip";
+    std::string SupplyNodeName;
+    std::string ReturnNodeName;
+    int TotNumEquip = 1;
+    int EquipIndex = 1; // Equipment index
+    DataZoneEquipment::ZoneEquipType zoneEquipType;
+
+    state->dataLoopNodes->NodeID.allocate(2);
+    state->dataLoopNodes->Node.allocate(2);
+    state->dataLoopNodes->NodeID(1) = "SupplyNode";
+    state->dataLoopNodes->NodeID(2) = "ReturnNode";
+
+    state->dataHVACVarRefFlow->GetVRFInputFlag = false;
+    state->dataHVACVarRefFlow->VRFTU.allocate(1);
+    state->dataHVACVarRefFlow->VRFTU(1).VRFTUOutletNodeNum = 1;
+
+    zoneEquipType = DataZoneEquipment::ZoneEquipType::VariableRefrigerantFlowTerminal;
+    state->dataHVACVarRefFlow->NumVRFTU = 1;
+    check = CheckEquipName(*state, EquipName, SupplyNodeName, ReturnNodeName, EquipIndex, zoneEquipType);
+    if (check) {
```

Comment:
All of these `if (check)` blocks should be replaced with `EXPECT_TRUE(check)`. As it stands if the check fails the unit test will pass, because the other `EXPECT_EQ`s will not execute.

Reply:
@lgu1234 this definitely needs to be changed.  Do you see what is happening?  If `check` comes back `false`, then you simply won't execute the `EXPECT_EQ` comparisons.  So the test would pass.  Instead, as @mjwitte says, you should convert:

```
    if (check) {
        EXPECT_EQ("SupplyNode1", SupplyNodeName);
    }
```

to:

```
    EXPECT_TRUE(check);
    EXPECT_EQ("SupplyNode1", SupplyNodeName);
```